File: ProiectMpp/ProiectMpp/Repository/AbstractDatabaseRepository.cs
using ProiectMpp.Domain;
using log4net;

namespace ProiectMpp.Repository
{
    abstract public class AbstractDatabaseRepository<TId, TE> : IRepository<TId, TE> where TE : Entity<TId>
    {
        protected IDictionary<TId, TE> Data;
        protected static readonly ILog Log = LogManager.GetLogger(typeof(PlayerDBRepository));
        protected readonly string ConnectionString;

        public AbstractDatabaseRepository(string connectionString)
        {
            ConnectionString = connectionString;
            Data = new Dictionary<TId, TE>();
        }

        abstract public TE? FindOne(TId id);

        public IDictionary<TId, TE> FindAll()
        { 
            return Data;
        }

        abstract protected void Load();
        abstract public TE? Save(TE entity);
        abstract public TE? Delete(TId id);
        abstract public TE? Update(TE entity);
    }
}

File: ProiectSemestru/src/main/java/project/moto/Repository/AbstractDatabaseRepository.java
package project.moto.Repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import project.moto.Domain.Entity;
import project.moto.Utils.JdbcUtils;

import java.util.*;

abstract public class AbstractDatabaseRepository<ID, E extends Entity<ID>> implements Repository<ID,E> {
    protected final JdbcUtils dbUtils;
    protected static final Logger logger = LogManager.getLogger();
    protected Map<ID,E> data;

    public AbstractDatabaseRepository(Properties props) {
        this.dbUtils = new JdbcUtils(props);
        this.data = new HashMap<>();
        Load();
    }
    public Map<ID,E> findAll(){
        return data;
    }

    abstract public Optional<E> findOne(ID id);
    abstract public Optional<E> save(E entity);
    abstract public Optional<E> delete(ID id);
    abstract public Optional<E> update(E entity);
    abstract protected void Load();

}
